api/client.go

```diff
@@ -297,17 +305,24 @@ func (c Client) DeleteProvider(id uid.ID) error {
 	return delete(c, fmt.Sprintf("/api/providers/%s", id), Query{})
 }
 
-func (c Client) ListGrants(req ListGrantsRequest) (*ListResponse[Grant], error) {
-	return get[ListResponse[Grant]](c, "/api/grants", Query{
-		"user":          {req.User.String()},
-		"group":         {req.Group.String()},
-		"resource":      {req.Resource},
-		"destination":   {req.Destination},
-		"privilege":     {req.Privilege},
-		"showInherited": {strconv.FormatBool(req.ShowInherited)},
-		"showSystem":    {strconv.FormatBool(req.ShowSystem)},
-		"page":          {strconv.Itoa(req.Page)}, "limit": {strconv.Itoa(req.Limit)},
-	})
+func (c Client) ListGrants(ctx context.Context, req ListGrantsRequest) (*ListResponse[Grant], error) {
+	q := Query{
+		"user":            {req.User.String()},
+		"group":           {req.Group.String()},
+		"resource":        {req.Resource},
+		"destination":     {req.Destination},
+		"privilege":       {req.Privilege},
+		"showInherited":   {strconv.FormatBool(req.ShowInherited)},
+		"showSystem":      {strconv.FormatBool(req.ShowSystem)},
+		"page":            {strconv.Itoa(req.Page)},
+		"limit":           {strconv.Itoa(req.Limit)},
+		"lastUpdateIndex": {strconv.FormatInt(req.LastUpdateIndex, 10)},
+	}
+	return request[EmptyRequest, ListResponse[Grant]](ctx, c, http.MethodGet, "/api/grants", q, nil)
+}
+
+func (c Client) ListGrantsOld(req ListGrantsRequest) (*ListResponse[Grant], error) {
```

Comment:
Is this just left here to clean-up in a different PR?

Reply:
Ya, this was temporary so that I didn't have to change all the list methods. I ended up doing that in e44127f138b450d1322479d29cfc8a9eb24ee69c. I've rebased this PR so this `ListGrantsOld` is gone now.